### PR TITLE
fix analysis errors

### PR DESCRIPTION
**Pull request by devoncarew**
- fix ~a dozen analysis issues related to `show` in exports
- fix lots of missing type annotations

These were found w/ the latest 1.17.0-dev sdk.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/debug.dart [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import 'package:flutter/painting.dart';
 import 'package:flutter/rendering.dart';
 
-export 'package:flutter/services.dart' show debugPrint;
+export 'package:flutter/services.dart';
```
Comment (abarth) [inline packages/flutter/lib/src/rendering/debug.dart]: I don't understand this change.  Previously we were exporting only a single name from services.dart.  Now we're exporting the whole thing.

Comment (devoncarew) [inline packages/flutter/lib/src/rendering/debug.dart]: Oops, yeah. This symbol no longer exists in `flutter/services.dart` and the latest analyzer complains about that for exports. I should have removed the whole line.

**Comment by devoncarew** [comment]
Updated to remove `exports` no longer necessary exports and to fix a test.

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
@abarth at some point we're going to have to go through and re-export names that we think should be re-exported.

**Comment by abarth** [comment]
I'm not sure I understand.  I think this lint is catching no-op exports (i.e., of names that don't exist).

**Comment by Hixie** [comment]
Yeah but in same cases they don't exist because they were renamed or moved. For example the debugPrint one is now in foundation instead of services. I'm sure there's other cases where over time we're bit-rotting our lists of exports, like where there's things we're not re-exporting but should be.

**Comment by abarth** [comment]
Ah, makes sense.  Hopefully this new lint will help us not rot more.

**Comment by devoncarew** [comment]
Landing; I did change the export of `debugPrint` from services to foundation.

**Pull request merged**